### PR TITLE
Filter paths that run CI

### LINKED ISSUE #221: Add rule to trigger github ci only when code files change
**Is your feature request related to a problem? Please describe.**
Currently github ci triggers for every commit even if it means changes in example and README files. This can be improved by only triggering ci on code file changes.

**Describe the solution you'd like**
This can be fixed by adding `rules` to `haskell-ci.yml`. An example take from this [SO answer](https://stackoverflow.com/questions/40712275/gitlab-ci-how-to-trigger-a-build-only-if-changes-happen-on-particular-set-of-f) is given below.

```
  rules:
    - changes:
      - /*.{java, py} # ... or whatever your file extension is
```

**Describe alternatives you've considered**
N/A

**Additional context**
Currently there are no end to end tests for examples (they might not be relevant) right now. When tests for examples are added they can also be added to changed files rules.

### PR DESCRIPTION
**Pull request by xsebek**
- only build and test when `.hs`, `.cabal` or `stack.yaml` files change
- closes #221

### CODE REVIEW

**Comment by xsebek** [comment]
Cool, it seems like the rules apply to the PR itself :slightly_smiling_face:

File: .github/workflows/haskell-ci.yml [hunk 1]
```diff
@@ -14,8 +16,17 @@
...
-  - push
-  - pull_request
+  push: 
+    paths:
+      - '**.hs'
+      - '**.cabal'
+      - 'stack.yaml'
+  # reuse the same filter for pull-requests
+  pull_request:
+    paths:
```
Comment (xsebek) [inline .github/workflows/haskell-ci.yml:26]: Surely we could avoid this duplicity with YAML anchors, right? [Well, no.](https://github.community/t/support-for-yaml-anchors/16128) :disappointed:

File: .github/workflows/haskell-ci.yml [hunk 2]
```diff
@@ -14,8 +16,17 @@
 #
 name: Haskell-CI
 on:
-  - push
-  - pull_request
+  push: 
+    paths:
+      - '**.hs'
+      - '**.cabal'
+      - 'stack.yaml'
```
Comment (TristanCacqueray) [inline .github/workflows/haskell-ci.yml:23]: Should it also keep `**.sw` and `cabal.project` ? And what about `**.yaml` for static tool configs (e.g. fourmolu.yaml) and entities if they are used in test?

Comment (TristanCacqueray) [reply .github/workflows/haskell-ci.yml:23]: Or is there a `irrelevant-path` option we could set to `**.md` instead?

Comment (xsebek) [reply .github/workflows/haskell-ci.yml:23]: There is, but I prefer adding .sw and cabal as I plan to add editor support which would waste Haskell CI.

Comment (xsebek) [reply .github/workflows/haskell-ci.yml:23]: Not sure about .yaml, maybe “data/**.yaml” and “*.yaml” but not “**.yaml” would work?

**Review by TristanCacqueray — APPROVED** [review]
Thanks!

**Review by byorgey — APPROVED** [review]
I guess since you requested my review it won't actually merge it until I approve?

**Comment by xsebek** [comment]
@byorgey Sorry to bother you, I shall try this again then. 😅

**Comment by xsebek** [comment]
So the issue is that mergify needs the jobs to be successful. I will try to find a good solution, maybe I could make a dummy job with paths-ignore that would always pass and then use mergify `or`? 😕

**Comment by xsebek** [comment]
> Configuration changed. This pull request must be merged manually 

Right, hopefully it works :worried:

**Pull request merged**